src/rgw/rgw_bucket_snap_types.h

```diff
@@ -0,0 +1,171 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#pragma once
+
+#include <string>
+#include <map>
+#include "include/types.h"
+#include "common/Formatter.h"
+#include "common/ceph_time.h"
+#include "common/ceph_json.h"
+#include "rgw_xml.h"
+
+
+
+struct rgw_bucket_snap_id {
+  static constexpr uint64_t SNAP_UNDEFINED = (uint64_t)0;
```

Comment:
prefer assignment that doesn't use cast syntax;  e.g., = 0ULL ;  I'm not sure even a type suffix is needed in the assignment needed at all though?